src/foam/nanos/u2/navigation/SignUp.js

```diff
@@ -67,150 +68,46 @@ foam.CLASS({
         {
-          args: ['jobTitle'],
+          args: ['userName'],
           predicateFactory: function(e) {
-            return e.NEQ(foam.nanos.u2.navigation.SignUp.JOB_TITLE, '');
+            return e.REG_EXP(
+              foam.nanos.u2.navigation.SignUp.USER_NAME,
+              /^[^\s\/]+$/);
           },
-          errorMessage: 'VALIDATION_ERR_TEXT'
+          errorString: 'Please enter username'
         }
       ],
-      required: true
-    },
-    {
-      class: 'PhoneNumber',
-      name: 'phone',
-      required: true
-    },
-    {
-      class: 'String',
-      name: 'organization',
-      label: 'Company Name',
-      view: {
-        class: 'foam.u2.TextField',
-        placeholder: 'ABC Company'
-      },
-      visibility: function(disableCompanyName_) {
-        return disableCompanyName_ ? foam.u2.DisplayMode.DISABLED : foam.u2.DisplayMode.RW;
-      },
-      required: true
-    },
-    {
-      class: 'Reference',
-      targetDAOKey: 'countryDAO',
-      name: 'countryId',
-      label: 'Country',
-      of: 'foam.nanos.auth.Country',
-      documentation: 'Country address.',
-      view: function(_, X) {
-        var E = foam.mlang.Expressions.create();
-        choices = X.data.slot(function(countryChoices_) {
-          if ( ! countryChoices_ || countryChoices_.length == 0 ) return X.countryDAO;
-          return X.countryDAO.where(E.IN(X.data.Country.ID, countryChoices_));
-        });
-        return foam.u2.view.ChoiceView.create({
-          placeholder: X.data.SELECTION_TEXT,
-          objToChoice: function(a) {
-            return [a.id, a.name];
-          },
-          dao$: choices
-        }, X);
-      },
       required: true,
-    },
-    {
-      class: 'String',
-      name: 'userName',
-      label: 'Username',
-      view: {
-        class: 'foam.u2.TextField',
-        placeholder: 'example123'
-      },
-      //TODO: uncomment when integrating
-      // validationPredicates: [
-      //   {
-      //     args: ['userName'],
-      //     predicateFactory: function(e) {
-      //       return e.REG_EXP(
-      //         foam.nanos.u2.navigation.SignUp.USER_NAME,
-      //         /^[^\s\/]+$/);
-      //     },
-      //     errorString: 'Please enter username'
-      //   }
-      // ],
-      //TODO: set to true when integrating
-      required: false,
-      //TODO: set to false when integrating
-      hidden: true
-    },
-    {
-      class: 'EMail',
-      name: 'email',
-      view: {
-        class: 'foam.u2.TextField',
-        placeholder: 'example@example.com'
-      },
-      visibility: function(disableEmail_) {
-        return disableEmail_ ?
-          foam.u2.DisplayMode.DISABLED : foam.u2.DisplayMode.RW;
-      },
-      required: true
+      hidden: false
     },
     {
       class: 'Password',
       name: 'desiredPassword',
       label: 'Password',
       view: {
-        class: 'foam.u2.view.PasswordView',
+        class: 'net.nanopay.ui.NewPasswordView',
```

Comment:
nanopay file can't be here - I have a pr that moves this but for now just use the current foam PasswordView

Reply:
Will undo. I could also add a register call in Controller for this, no?

Reply:
If you wanted to replace all references of PasswordView to NewPasswordView then ya - that'd do it.

Reply:
If the NewPasswordView is more full featured - then just move it to FOAM.

Reply:
The issue @jlhughes was the NewPasswordView has the entropy api.

I had written a new entropy based on our password policy and all was going well till we had to pick the group for the password policy - and then we discussed writing something in the index/html ... and I got lost and left it - at the time it was more of a clean up then a necessity. But now its a necessity. 
I'll go back and figure this out.